docs/scarpet/api/BlocksAndWorldAccess.md

```diff
@@ -246,6 +246,57 @@ NB: It can thunder without there being a thunderstorm, there has to be both rain
 
 With two args, sets the weather to `type` for `ticks` ticks.
 
+### `track_changes()`
+Keep track of any changes to the block states of blocks made by your scarpet script so they can be undone later if neccesary using `undo_changes()`. All changes tracked are undone when the script unloads, or if the server closes (unless you close a server with the x button, or it crashes). Does nothing if changes are already being tracked.
+
+Returns a boolean for whether changes were being tracked before
+
+Only works in globally scoped apps
```

Comment:
I don't see a reason why this needs to be a thing. Explain

Reply:
For the returns thing, I didn't really see a reason not to, I'd be happy to remove it

For the globally scoped thing, I just couldn't figure out how to get it to work right for player scoped apps, and I figured it didn't matter much anyways, since the new features are more world focused. Also, if it would work in player scoped apps, it'd probably work such that each player would keep track of changes individually and undo individually which I think would be strange and not very useful behavior.

Reply:
consider a usecase - a veinminer. Its a typical player scoped usecase. All modified blocks are stored with the player script host, so not affecting other players. Makes perfect sense and the way you have implemented it - with storing the undo data in players' script host the support for multiple users comes automatic, doesn't it?

Reply:
Well sort of, I noticed while writing the example that running the commit command wouldn't work properly, and the changes would undo anyways, and I thought it was some sort of strange side effect with how player scoping was implemented. Rereading the documentation, it sounds like this is because the data wasn't being copied correctly. Ima try and give it another go.

Reply:
Ok, it seems to be working now for player scope